test/e2e/storage/vsphere/vsphere_zone_support.go

```diff
@@ -0,0 +1,378 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+const (
+	VsanDatastore1  = "vsanDatastore"
+	VsanDatastore2  = "vsanDatastore (1)"
+	CompatPolicy    = "compatpolicy"
+	NonCompatPolicy = "noncompatpolicy"
+	ZoneA           = "zone-a"
+	ZoneB           = "zone-b"
+	ZoneC           = "zone-c"
+	ZoneD           = "zone-d"
+)
+
+/*
+   Test to verify multi-zone support for dynamic volume provisioning in kubernetes.
+   The test environment is illustrated below:
+
+   datacenter
+   	--->cluster-vsan-1 (zone-a)          	 ____________________	 _________________
+   		--->host-1 : master             |                    |	|		  |
+   		--->host-2 : node1      	|   vsanDatastore    |	|		  |
+   		--->host-3 : node2      	|____________________|	|		  |
+									|		  |
+									|   sharedVmfs-0  |
+	--->cluster-vsan-2 (zone-b) 	  	 ____________________	|		  |
+   		--->host-4 : node3      	|                    |	|		  |
+   		--->host-5 : node4      	|  vsanDatastore (1) |	|		  |
+   		--->host-6       		|____________________|	|_________________|
+
+	--->cluster-3 (zone-c)			 ________________
+		--->host-7 : node5       	|                |
+						| localDatastore |
+						|________________|
+						 ____________________
+        --->host-8 (zone-c) : node6          	|		     |
+						| localDatastore (1) |
+        					|____________________|
+
+
+	Testbed description :
+	1. cluster-vsan-1 is tagged with zone-a. So, vsanDatastore inherits zone-a since all the hosts under zone-a have vsanDatastore mounted on them.
```

Comment:
Is there any automation for creating the testbed categories/tags, tagging resources, etc., or is this done manually?

Reply:
Yes. The entire testing process including tagging, policy creation etc.. is automated through a jenkins job.

Reply:
Can you publish the jenkins job config / script somewhere for folks who don't have access to this jenkins instance?

Reply:
Updated PR description with the jenkins automation script link.